wgpu/examples/shadow/main.rs

```diff
@@ -292,9 +292,11 @@ impl framework::Example for Example {
 
         let entity_uniform_size = mem::size_of::<EntityUniforms>() as wgpu::BufferAddress;
         let num_entities = 1 + cube_descs.len() as wgpu::BufferAddress;
-        let uniform_alignment =
+        let mut uniform_alignment =
             device.limits().min_uniform_buffer_offset_alignment as wgpu::BufferAddress;
-        assert!(entity_uniform_size <= uniform_alignment);
+        // Make the `uniform_alignment` >= `entity_uniform_size` and aligned to `min_uniform_buffer_offset_alignment`.
+        uniform_alignment *=
+            (entity_uniform_size as f32 / uniform_alignment as f32).ceil() as wgpu::BufferAddress;
```

Comment:
Let's not involve floating-point arithmetic here. Instead:
```rust
let uniform_stride = {
  let alignment = device.limits().min_uniform_buffer_offset_alignment;
  let rem = entity_uniform_size % alignment;
  if rem != 0 {
    entity_uniform_size + alignment - rem
  } else {
    entity_uniform_size
  }
};
```

Reply:
If I'm not mistaken, `entity_uniform_size` is an integer multiple of `alignment` and greater than 1, `uniform_stride` result will be wrong?

Reply:
> If I'm not mistaken, when entity_uniform_size is an integer multiple of alignment and greater than 1, uniform_stride result will be wrong?

I don't think so. In this case `rem == 0`, which means `uniform_stride == entity_uniform_size`, which is what we want